common/dynamicconfig/constants.go

```diff
@@ -1429,6 +1429,12 @@ timeout timer when execution timeout is specified when starting a workflow.`,
 		true,
 		`EnableUpdateWorkflowModeIgnoreCurrent controls whether to enable the new logic for updating closed workflow execution
 by mutation using UpdateWorkflowModeIgnoreCurrent`,
+	)
+	EnableReturnNewWorkflowTaskUnconditionally = NewGlobalBoolSetting(
+		"history.enableReturnNewWorkflowTaskUnconditionally",
+		false,
```

Comment:
I'd probably enable it by default for development (`config/dynamicconfig`) and all functional tests (`test/testcore/dynamic_config_overrides.go`)